cmd/bnsd/x/username/username_test.go

```diff
@@ -59,25 +47,19 @@ func TestUsername(t *testing.T) {
 			WantName:   "foo",
 			WantDomain: "",
 		},
-		"missing name": {
-			Raw:        "*iov",
-			WantErr:    errors.ErrInput,
-			WantName:   "",
-			WantDomain: "iov",
-		},
 		"missing separator": {
 			Raw:        "xyz",
 			WantErr:    errors.ErrInput,
 			WantName:   "",
 			WantDomain: "",
 		},
 		"invalid characters (emoji)": {
-			Raw:        "😈*😀",
+			Raw:        "😈*iov",
```

Comment:
pp: Can you add a test for `*foo*iov` or something similar which uses the separator char inside the name part, please? This char is reserved and a test would help to cover and enforce this contract.